dataservices/service/server/config.go

```diff
@@ -58,3 +58,17 @@ func (c *Config) Validate() error {
 	}
 	return nil
 }
+
+func (c *Config) Clone() *Config {
```

Comment:
You create the config, pass the config into some object (MongoStore) that remembers the config, then you change the config. You've inadvertently changed the MongoStore by changing the config *after* giving it to the MongoStore. I ran into this while writing the `migrate_gid_to_uid` tool because I was using the same config (with the mongo address) for multiple mongo stores (user, seagull, data).